src/app/shared/money-input/money-input.component.ts

```diff
@@ -0,0 +1,54 @@
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { Component, forwardRef } from '@angular/core';
+
+@Component({
+  selector: 'app-money-input',
+  templateUrl: './money-input.component.html',
+  styleUrls: ['./money-input.component.css'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => MoneyInputComponent),
+      multi: true,
+    },
+  ],
+})
+export class MoneyInputComponent implements ControlValueAccessor {
+  private _amount: number = 0;
```

Comment:
If you set initial value you shouldn't specify a type

Reply:
Do not use underscore if you set `private ` access modifier

Reply:
done